lms/js_config_types.py

```diff
@@ -63,7 +64,7 @@ class AutoGradingGrade(TypedDict):
     last_grade: float | None
     """Last grade that was succefully sync to the LMS."""
 
-    last_grade_date: str | None
+    last_grade_date: datetime | None
```

Comment:
This is an unintended consequence of this change.

Whie this tries to model the same types as config.ts this is obviously python so I reckon is fine to have this differences.

Reply:
Yeah, mypy does not know about how this is serialized anymore, so I guess we have to live with it.

The API models (and the API definition itself) should be documented separately, specially since it is becoming quite big.

We should consider adding some OAS/swagger definition at some point.